network/p2p/topic_validator_test.go

```diff
@@ -371,8 +361,73 @@ func TestAuthorizedSenderValidator_Ejected(t *testing.T) {
 
 	unittest.RequireReturnsBefore(t, wg.Wait, 5*time.Second, "could not receive message on time")
 
-	// expecting 1 warn calls for each rejected message from ejected node
-	require.Equalf(t, uint64(1), hookCalls, "expected 1 warning to be logged")
+	// expecting ErrSenderEjected to be logged
+	require.Regexp(t, validator.ErrSenderEjected, hook.Logs())
+}
+
+// TestAuthorizedSenderValidator_Ejected tests that the authorized sender validator rejects messages from nodes that should only receive on the channel
+func TestAuthorizedSenderValidator_ReceiveOnly(t *testing.T) {
+	sporkId := unittest.IdentifierFixture()
+	identity1, privateKey1 := unittest.IdentityWithNetworkingKeyFixture(unittest.WithRole(flow.RoleConsensus))
+	sn1 := createNode(t, identity1.NodeID, privateKey1, sporkId)
```

Comment:
`createNode` is similar to the already existing [`nodeFixture`](https://github.com/onflow/flow-go/blob/58d3927376643892c39138ce079b87a43d2f548d/network/p2p/fixture_test.go#L88), which resides in the same `p2p_test` package. The `nodeFixture` is a more realistic fixture as it configs the `libp2p` node closely to what we build on a Flow node. Please use that instead and remove the `createNode` one, as having redundant fixtures with similar functionalities degrade the integrity of our tests over the time and also makes the test fixture harder to maintain. Thanks! 

Note-1: most probably you need to add a [`withRole`](https://github.com/onflow/flow-go/blob/58d3927376643892c39138ce079b87a43d2f548d/network/p2p/fixture_test.go#L48) option to `nodeFixture` similar to the other options that we already have. This allows the test to instantiate the libp2p nodes with a specific role. 

Note-2: the `nodeID` parameter of `createNode` is unused.